src/Components/PagesBody/ReservationsPage/ReservationsPage.tsx

```diff
@@ -1,49 +1,91 @@
-import { Box, Tab, Tabs } from '@mui/material';
+import { Box, Tab, Tabs, Typography } from '@mui/material';
+import ReservationCard, {
+  ReservationCardTypes,
+} from '../../Organisms/ReservationCard/ReservationCard';
+import { getDocs, orderBy, query, where } from 'firebase/firestore';
+import {
+  innerContainerStyle,
+  reservationCardsBoxStyle,
+  reservationWrapperStyle,
+  tabStyle,
+} from './ReservationPageStyles';
+import { useEffect, useState } from 'react';
 
-import ReservationCard from '../../Organisms/ReservationCard/ReservationCard';
 import TabPanel from '../../Organisms/ReservationCard/TabPanel';
-import { useState } from 'react';
+import { reservationsRef } from '../../../Firebase/firebase';
+
+//mock for userId
+const userId = 'Pk1ORdo4QSDzwCet2nT7';
 
 const ReservationsPage = () => {
   const [value, setValue] = useState(0);
+  const [reservations, setReservations] = useState<
+    ReservationCardTypes[] | null
+  >(null);
+
+  useEffect(() => {
+    async function fetchReservations() {
+      try {
+        const snapshot = await getDocs(
+          query(
+            reservationsRef,
+            where('userId', '==', userId),
+            orderBy('serviceDate', 'desc'),
+          ),
+        );
+        if (!snapshot) {
+          setReservations(null);
+        } else {
+          const reservationsArr: ReservationCardTypes[] = [];
+          snapshot.forEach((reservation) => {
+            const reservationData = reservation.data();
+            const reservationObj = {
+              serviceName: reservationData.serviceName as string,
+              serviceDate: reservationData.serviceDate.toDate() as Date,
+            };
+            reservationsArr.push(reservationObj);
+          });
+
+          setReservations(reservationsArr);
+        }
+      } catch (error) {
+        alert('Oops, coś poszło nie tak, odśwież stronę');
```

Comment:
Maybe we should avoid leaving alerts ? There will be error handling instead

Reply:
for me it depends, how error handling will work. A that current situation I think is better to show info connected to error in fetching data, than crash application by throwing error